Fix RSA premaster

mattcaswell opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] CLA is signed

##### Description of change
<!-- Provide a description of the changes.

If it fixes a github issue, add Fixes #XXXX.
-->

ekasper commented on ssl/statem/statem_lib.c:
```diff
@@ -1049,8 +1049,8 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
              */
-            if ((int)candidate_vers > s->client_version)
-                s->client_version = candidate_vers;
+            /*if ((int)candidate_vers > s->client_version)
```
So the effect of this change is that the client's highest supported version from the extension is used to choose the server version, but this value is not recorded anywhere?

Can you explain a bit more why that is correct w.r.t. surrounding code? Does nothing else assume that `s->client_version` is the highest supported client version? Perhaps stick a comment in `ssl_locl.h` reminding us that this is the `legacy_version` field in TLS 1.3?

mattcaswell commented:
I pushed an update to this. I removed the code rather than commenting it out. I also tweaked the code around adding legacy_version to the ClientHello to make it a bit neater. I also updated the comment in ssl_locl.h for the client_version field.

mattcaswell commented:
Pushed. Thanks.
